src/db.rs

```diff
@@ -56,6 +57,47 @@ const KEEP_LOGS: usize = 16;
 /// Value is just a vector of bytes. Value sizes up to 4Gb are allowed.
 pub type Value = Vec<u8>;
 
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct ValuePtr(Arc<Value>);
```

Comment:
+ 1 to @arkpar I got confused with the `Ptr`, thinking the wrapped value is maybe `*const Value`.

Also, what about just doing `pub type ValuePtr = Arc<Value>` to get for free everything that is already implented for `Arc<Value>` ? I am not sure the "new type" pattern brings something here.

Reply:
Sure I can rename it.  I use pointer because in the [rust book](https://doc.rust-lang.org/stable/book/ch15-00-smart-pointers.html), `Rc`, `Box` etc are introduced as `smart pointer`

Reply:
> what about just doing `pub type ValuePtr = Arc<Value>`

It's used as key type in `HashMap` and `BTreeMap`, but it cannot be queried by `&[u8]` for free

Not working ([playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=2066f2b7f006da3b011d3bc3c8c7860a))
```rust
use std::collections::HashSet;
use std::sync::Arc;

type RcValue = Arc<Vec<u8>>;

fn main() {
    let mut set: HashSet<RcValue> = HashSet::new();
    set.insert(vec![1, 2, 3].into());
    println!("{}", set.contains(&[1, 2, 3][..]));
}
```

Working ([playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=fecb6b574037c56245bba463d7453a9c))
```rust
use std::borrow::Borrow;
use std::collections::HashSet;
use std::sync::Arc;

#[derive(Eq, Hash, PartialEq)]
struct RcValue(Arc<Vec<u8>>);


impl Borrow<[u8]> for RcValue {
	fn borrow(&self) -> &[u8] {
		(&*self.0).borrow()
	}
}

impl From<Vec<u8>> for RcValue {
    fn from(value: Vec<u8>) -> Self {
        Self(value.into())
    }
}

fn main() {
    let mut set: HashSet<RcValue> = HashSet::new();
    set.insert(vec![1, 2, 3].into());
    println!("{}", set.contains(&[1, 2, 3][..]));
}
```

Reply:
Makes sense.